### DISCUSSION TITLE
Simple Job DLL with Quartz.Server

### DISCUSSION THREAD

**Discussion by Scimiot** [quartznet/quartznet#2147] [Q&A]
Hi All,

I'm struggling with the simplest of tasks. I read all the documentation, looked at the examples but still I cannot figure out where is the mistake I'm making.

I'm running Quartz through Quartz.Server.exe (3.6.2 .net6 for example) and I would like to make it run a Job I wrote in a simple .net6 DLL. This projects contains just one class that is the following:

```cs
using log4net;
using Quartz;

namespace QuartzJobCore6
{
    public class FirstJob : IJob
    {
        private static readonly ILog logger = LogManager.GetLogger(typeof(FirstJob));

        public async Task Execute(IJobExecutionContext context)
        {
            logger.Info("SampleJob running...");
            await Task.Delay(TimeSpan.FromSeconds(5));
            logger.Info("SampleJob run finished.");
        }
    }
}
```

After compiling, I copy the DLL in the same folder of Quartz.Server.exe and I modify quartz_jobs.xml by adding the following XML portion instead of the default one:

```xml
    <job>
      <name>FirstJob</name>
      <group>sampleGroup</group>
      <description>Sample job for Quartz Server</description>
      <job-type>QuartzJobCore6.FirstJob, QuartzJobCore6</job-type>
      <durable>true</durable>
      <recover>false</recover>
      <job-data-map>
        <entry>
          <key>key1</key>
          <value>value1</value>
        </entry>
        <entry>
          <key>key2</key>
          <value>value2</value>
        </entry>
      </job-data-map>
    </job>
    <trigger>
      <simple>
        <name>sampleSimpleTrigger</name>
        <group>sampleSimpleGroup</group>
        <description>Simple trigger to simply fire sample job</description>
        <job-name>FirstJob</job-name>
        <job-group>sampleGroup</job-group>
        <misfire-instruction>SmartPolicy</misfire-instruction>
        <repeat-count>-1</repeat-count>
        <repeat-interval>10000</repeat-interval>
      </simple>
    </trigger>
```

Then, I run Quartz.Server.exe from command line but I alwasy get this error, amongst the standard output:

```cmd
2023-09-28 14:14:10,896 [1] INFO  Quartz.Server.QuartzServer - Scheduler started successfully
The QuartzServer service is now running, press Control+C to exit.
2023-09-28 14:14:10,978 [7] ERROR Quartz.Plugin.Xml.XMLSchedulingDataProcessorPlugin - Could not schedule jobs and triggers from file ~/quartz_jobs.xml: Could not load type 'QuartzJobCore6.FirstJob, QuartzJobCore6'
```

I tried many combinations of namespace and job name, all leading to the same result. If I do the whole thing using .NET Framework 4.8 it works perfectly. My best guess is that I make some mistakes while compiling the DLL, but I really cannot figure out why this happens.

Any clues?

The Visual Studio solution can be found at the following address: https://github.com/Scimiot/Simple-Quartz-.NET-Core-6-Job

**Comment by jafin** [comment]
Is the compiled assembly named QuartzJobCore6?

**Reply by Scimiot** [reply]
Yes. Visual Studio creates a DLL in the Release folder, named QuartzJobCore6.dll.

**Comment by lahma** [comment]
Is log4net.dll also present in the folder? Some missing dependency could also cause trouble.

**Reply by Scimiot** [reply]
Thanks for your response. Yes, it is present. I've also used the same version as the one available in the Quartz.Server folder, just to be 100% sure.

**Reply by Scimiot** [reply]
As update, I tried to recreate the same setup on a brand new virtual machine. In this way I'm excluding issues related with the specific dev machine I'm working with.

These are the operations I've done on the brand new virtual machine:

- net6 DLL with the same source code as before but recreated from scratch.
- Quartz.Server downloaded from github.
- Same quartz_jobs.xml

The output is still the same.

**Reply by lahma** [reply]
Maybe you could create a public github repository demonstrating the problem?

**Reply by Scimiot** [reply]
I published the project at the following address: https://github.com/Scimiot/Simple-Quartz-.NET-Core-6-Job

Thanks!

**Reply by lahma** [reply]
That doesn't look like anything that could be easily run to see the problem that you are facing?

**Reply by Scimiot** [reply]
I've updated the repository content: now there is a copy of Quartz.Server folder (3.7.0) with the compiled DLL (QuartzJobCore6.dll) and quartz_jobs.xml configured correctly. 

To reproduce the error it's sufficient to launch Quartz.Server.exe from cmd.

Thanks again!

**Reply by lahma** [reply]
I think that the GetType that Quartz uses cannot load the type as the assembly hasn't been loaded automatically, so probing won't currently consider anything that hasn't been referenced by the server.

**Reply by Scimiot** [reply]
An additional thought that might be helpful: using Quartz.Server folder as net462 version with the compiled DLL (QuartzJobCore6.dll)  targeting Framework 4.8 works fine. The job is loaded and works fine.

Might be a behavior of GetType while Quartz targets net6 instead of net462?

**Reply by Scimiot** [reply]
@lahma Any suggestions/possible directions?

**Reply by lahma** [reply]
You could compile the server yourself and have reference to your dll. The server is mostly a sample and people might want to modify it anyway.
